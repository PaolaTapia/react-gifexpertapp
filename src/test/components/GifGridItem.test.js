import React from 'react'; 
import { shallow } from 'enzyme';

import GifGridItem from '../../components/GifGridItem';

describe('Prueba de GifGridItem', ()=> {

     const title = 'un titulo'; 
     const url = 'https://localhost/imagen.jpg'; 
     const wrapper=shallow(<GifGridItem title={title} url={url}/> );
     
     test('debe mostrar el componente correctamente', () => {

          const wrapper = shallow(
               <GifGridItem title={title} url={url}/> 
               );
               
          expect(wrapper).toMatchSnapshot();
          
     }); 

     test('debe de tener un parrafo con el title', () => {

          const p=wrapper.find('p'); 
          
          expect(p.text().trim()).toBe(title); 
          
     }); 

     test('debe de tener la imagen igual al url y alt de los props', () => {

          const img= wrapper.find('img'); 

          expect(img.props().src).toBe(url); 
          expect(img.props().alt).toBe(title);           
     }); 
     test('debe tener la clase animate__animated animate__bounce ', () => {

          const div= wrapper.find('div'); 
          const className=div.prop('className'); 
          
          expect(className.includes('animate__bounce')).toBe(true); 
          
     }); 
     
     
})
