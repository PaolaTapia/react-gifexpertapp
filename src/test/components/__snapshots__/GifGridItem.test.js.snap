// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="animate__animated animate__bounce card"
>
  <img
    alt="un titulo"
    src="https://localhost/imagen.jpg"
  />
  <p>
     
    un titulo
  </p>
</div>
`;
